### DISCUSSION TITLE
NREL 5MW ITI Barge RAO calculations

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1368] [Q&A]
Dear everyone
I am studying the RAO of this paper(Investigation of Response Amplitude Operators for Floating Offshore Wind Turbines). I have a problem that is to calculate the RAO of six degrees of freedom of ITI Barge. I will calculate four kinds of wind cases:No wind, 8m/s, 11.4m/ and 20m/s. The results of the six degrees of freedom are shown in the following figure. I am not sure whether the calculation result is correct, because I am also vague about the hydrodynamic module settings and I refer to the settings of many forums. I also display the hydrodynamic settings in the figure below，Do all the diagrams look correct？Another question is, how can I be consistent with the coordinate display of the last graph, and how can the smoothness of the graph look consistent？
![image](https://user-images.githubusercontent.com/107483730/205481992-13e63a23-b751-43fc-833e-21a5c413a391.png)
![image](https://user-images.githubusercontent.com/107483730/205482023-2e2bd994-d208-47b8-b4d4-719e3bc1bfb7.png)
![image](https://user-images.githubusercontent.com/107483730/205483935-06c9de46-0bc6-4e45-8024-2bd87fdf13d8.png)

![image](https://user-images.githubusercontent.com/107483730/205482435-c2271a65-9688-4f88-ab30-6c80f186d048.png)
It can be seen from the figure that the surge natural frequency is 0.008Hz, and the amplitude and the natural frequency of surge at the rated wind speed is slightly higher. In no wind case, it is uncertain whether it is coupled with pitch (0.08401Hz), because this amplitude is small. There is a 0.08001Hz in front of it. In operational conditions, there is no coupling with the surge-heave or surge-pitch, and there is a 0.07601Hz in front of it. In the no wind case, the peak value of the surge is the smallest. A smaller peak is observed at about the first blade flap-wise natural frequency (0.6121 Hz).

![image](https://user-images.githubusercontent.com/107483730/205482680-27964ec8-de97-41ff-b2cb-e2718f2ca445.png)
The figure shows that there is excitation at sway natural frequency (0.008Hz) and roll natural frequency (0.084Hz), indicating sway-roll coupling. The amplitude of swing is very small. Because the sway motion is excited only by rotor torque and transverse aerodynamic loads brought about by rotor gyrocopics-induced yaw motion. Because there is no rotor load, the magnitude of no wind RAOs is very small compared with the operational conditions. At 8m/s, 11.4m/s and 20m/s wind speeds, the response under operational conditions is similar in magnitude, but slightly increases with the increase of wind speed, which is the result of higher aerodynamic torque. In addition, the effect of aerodynamic damping does not exist in this degree of freedom. In the case of a flexible turbine, a small peak can be seen at the bending frequency (0.6121 Hz) in the swing direction of the first coupled blade. However, the frequency amplitude of FAST RAOs on the side of the first tower is not seen.

![image](https://user-images.githubusercontent.com/107483730/205483039-3e237036-7074-44e1-9541-255454ef9865.png)
The natural frequency of the Heave Rao (0.088 Hz) is relatively good. This is because the heave response at the natural frequency is not affected by aerodynamic damping.

![image](https://user-images.githubusercontent.com/107483730/205483079-69d4857c-2902-402c-8b50-57fdd2ac74df.png)
In the figure, the roll-sway coupling is not seen. The roll blade flap wise binding natural frequency (0.6121 Hz) is seen.

![image](https://user-images.githubusercontent.com/107483730/205483148-2240ceee-94cc-4eed-91bd-8e0656da870c.png)
Pitch is not similar to surge, but is similar to roll. The pitch natural frequency is 0.084 Hz. Under operational conditions, RAOs have a better pitch natural frequency, but its amplitude is smaller than that the RAOs of no wind. This shows the effect of aerodynamic damping. Pitch is not coupled to surge. There is also no coupled to heave. The flexible turbine RAOs do not couple the frequencies of the first fore-aft tower frequency. I am not sure whether the frequency of the tower coincides with the pitch frequency. In addition, compared with the no wind case, the peak value at the pitch frequency shifts slightly in the wind case, which may be due to the additional aerodynamic stiffness or the stiffening of the mooring system to the aerodynamic-thrust-induced mean surge offset.

![image](https://user-images.githubusercontent.com/107483730/205483419-fab0604f-acdb-403f-a85b-c9baa192904e.png)
It can be seen that the natural frequency of YAW is 0.02Hz This is because the hydrodynamic yaw force is a zero cylindrical structure, and the main reason for this excitation should come from the aerodynamics of the rotor, especially the flexibility of the rotor. All raos have no excitation at the natural frequencies of surge and sway, which shows the coupling of yaw with pitch and roll. In addition, under operational conditions, all RAOs calculated by fast show an excitation close to 0.202 Hz, corresponding to 1p excitation from the rotor. This effect is more significant at rated and above wind speeds. The flexible turbine RAOs do not have the first tower side-side frequency, a and it is uncertain whether they overlap the frequency of pitch or roll.

![image](https://user-images.githubusercontent.com/107483730/205483901-c6544598-fcb2-48ca-96b5-42f502d4fd70.png)
![image](https://user-images.githubusercontent.com/107483730/205483905-eb545928-1088-4d98-8747-a60aa252b8ae.png)
![image](https://user-images.githubusercontent.com/107483730/205484055-0d41f88e-0959-40d0-ac7f-2b9d506781be.png)

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

I don't recall generating RAOs for the ITI Energy barge this way, so, I can't share results for you to compare.  But generally, your approach and results seem fine.  Are you using the ROA script and following the approach outlined in the following OpenFAST issue and associated forum posts: https://github.com/OpenFAST/openfast/issues/673?  Are you eliminating start-up transients before post-processing the results?

Regarding your HydroDyn settings, they look OK to me, although I am questioning why you are using `RdtnMod` = 2 (state space) rather than 1 (convolution).   The latter is more robust and the former is meant for linearization and depends on the accuracy of the state-space fit.

Regarding smoothing of the RAOs, the script linked above uses ensemble averaging to smooth the result.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
1. For this question, how do you actually eliminating start-up transients before post-processing the results? Sorry，I don't know how to operate. Could you tell me how to deal with it？
2. For your second question, I will modify RdtnMod=1 according to your suggestions, and I will briefly describe the changes in the results later.
3. The second and third pictures from the bottom are about Rao of TTDspFA and TTDspSS. From the results, they are coupled with pitch（0.08401Hz） and roll (0.08401Hz) respectively, but I'm not sure about the natural frequencies of the tower in these two pictures. They seem to be 0.5-0.6Hz. I get the natural frequencies of the tower from the free attenuation of the onshore wind turbine test as 0.323 (TTDspFA) and 0.322 (TTDspSS). Is the natural frequency of offshore wind turbine different from that of onshore wind turbine tower? So what do you think of this result?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Here are my responses:

Regarding (1), you can eliminate the start-up transient by setting `TStart` after the transients are sure to have been dissipated (requires some understanding of how long the start-up transients will last) or by post-processing (throw away the time series data that is output during the start-up transient).

Regarding (3), the tower natural frequency changes (increases) when mounting the tower on a floating platform.  I don't recall the exact tower frequency for the ITI Energy barge, but I would expect the frequency to fall between the land-based frequency and double that frequency.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/206075786-60d1f69a-4aea-44a8-9e74-5a16fe865498.png)
As shown in the picture, I understand that the start-up transients is about 30 seconds. Although I am not sure about this moment, I set TMAX=1000s, and whether I can take 400-1000 seconds of data, which I think should be stable. IF I calculate the RAO data, all of which take 400-1000s of data. For example, the heave RAO is the heave-wave cross-spectral density divided by the wave auto-spectral density. I take 400-1000s "Wave1Elev" data and 400-1000s "Heave" data. I'm not sure that's right.
Another question is, if I change the WaveMod to 1: regular (periodic) or 2: JONSWAP/Pierson Moskowitz spectrum (irregular), whether the "Wave1Eev" calculated by them separately can also be used as RAO calculation data, even though the forum said that the RAO script would function best if U was based on a white noise spectrum (WaveMod=3). And RAO script "nens = number of ensembles to average (i.e., 4 or 6)" , Is nens=4 generally selected?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Regarding the start-up transients, 30 s is typically sufficient for a land-based wind turbine.  For a floating wind turbine with low-frequency modes, e.g. in surge or pitch, it may take longer (as noted) unless proper initial conditions (based on the mean wind speed) are set in these DOFs as well.

Otherwise, I agree that you should use the same time period for outputting the response and wave elevation when computing ROAs.

Regular waves only excite the structure at one frequency, so, you can't get more than the RAO at a single frequency if you simulate with regular waves.  For irregular waves, the energy is concentrated around a given frequency band and negligible at lower and higher frequencies, so, you'd only be able to compute the RAO within this frequency band.  Moreover, the JONSWAP spectrum has more energy at some frequencies than others, so this could impact the contribution of nonlinearities in the response at different frequencies resulting in a skewed RAO.  This is why it is better to use white-noise waves to generate ROAs.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
As shown in the figure, the calculation result is different from the frequency result in Table 2. OC3-Hywind Platform and Tower Natural Frequencies  given in the paper( Investigation of Response Amplitude Operators for Floating Offshore Wind Turbines) . I had packed and compressed the dat，the fst files and calculation results xlsx, and attached OC3-RAO.zip at the bottom. I'm not sure where the setting is wrong, please correct my mistake.And I don't know how to be exactly the same as the picture in the paper. No matter from the trend or smoothness, I don't want the curve to be like a broken line. By the way, How do I set the working condition circled in the last picture？
![image](https://user-images.githubusercontent.com/107483730/206889457-e2ffe132-1056-4187-b2eb-66eb83873ded.png)
![image](https://user-images.githubusercontent.com/107483730/206889538-0a803aec-ebff-41ff-b5da-05c95bc9b841.png)
![image](https://user-images.githubusercontent.com/107483730/206889662-588fa07c-68e6-46f1-8fc8-ecf4af3f86de.png)
![image](https://user-images.githubusercontent.com/107483730/206889712-f8d1f895-a067-4d86-83bd-9af18641cc29.png)
![image](https://user-images.githubusercontent.com/107483730/206889752-e862ef64-7d10-47d5-92e6-d65971ccfd29.png)
![image](https://user-images.githubusercontent.com/107483730/206889786-0f1cab1b-7223-4b19-a587-827621662285.png)
![image](https://user-images.githubusercontent.com/107483730/207088150-f3187b55-a73f-4361-b19c-fad202b3d909.png)
[OC3-RAO.zip](https://github.com/OpenFAST/openfast/files/10201595/OC3-RAO.zip)
[OC3-result.xlsx](https://github.com/OpenFAST/openfast/files/10201604/OC3-result.xlsx)

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

I haven't reviewed your files, but overall your results look close to the paper you reference.  Your RAOs are generally aligned with the the published ROAs and the natural frequencies generally match Table 2.  Of course, there could be differences due to different wind conditions or model settings.  And you can use ensemble averaging to smooth the RAOs.  Do you have a specific concern?

Regarding the highlighted working conditions, these can be set as follows:

- `WavePkShp` in HydroDyn is equivalent to gamma
- The sub-surface current model in HydroDyn uses a 1/7th power law, so, set `CurrMod` = 1 with `CurrSSV0` = 0.5 m/s to enable this model
- The one-sided PSD implemented in HydroDyn for banded white noise is documented in the following topic on our forum: https://forums.nrel.gov/t/rao-oc3hywind-turbine/1753/14?u=jason.jonkman.  You can use this formula to determine what value of `WaveHs` to specify to get 1 m^2/Hz from `WvLowCOff` = 0.05 Hz to `WvHiCOff` = 0.25 Hz when using a white noise spectra (`WaveMod` = 3).  Please note that HydroDyn uses rad/s instead of Hz  

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/207253648-6c8bc5a3-5418-4efa-8236-47658417fe3c.png)
![image](https://user-images.githubusercontent.com/107483730/207268105-d42facf1-ebe4-44d6-a895-4ef80008b0e1.png)
![image](https://user-images.githubusercontent.com/107483730/207268223-2c41fedd-1880-443a-9f51-0f1aff6dd156.png)
As shown in the picture, the RAO of surge shows the natural frequency of surge is 0.007408, the second peak is 0.03704(I guess is pitch natural frequency ), and the third peak is 0.4741(I guess is TTDspFA natural frequency ). Although surge, pitch natural frequency are differences between I calculate and the paper given, the results should be acceptable. My concern is that I set the same condition(No wind,WaveMod=3,WaveHs=2,WaveTp=10,WvLowCOff=0,WvHiCOff=11, other is "default"). The picture in the paper does not appear the natural frequency of the tower(TTDspFA), and the RAO I calculated does not appear the natural frequency of the blade flap-wise(The paper shows 0.63Hz).

Another question is how do I set the angle between wind and wave? For example, 0 degrees. I think they are parallel. What if it is 60 degrees?

Can WaveHs and WaveTp be set at will? For example, WaveHs=2, 5, 13, 20, etc. WaveTp=10, 25, 30, etc. Or do we need formulas to calculate the corresponding relationship between WaveHs and WaveTp?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

> The picture in the paper does not appear the natural frequency of the tower(TTDspFA), and the RAO I calculated does not appear the natural frequency of the blade flap-wise(The paper shows 0.63Hz).

The surge response you show from the paper does not cover high enough frequencies to show the tower response.  However, you can see the tower response in other figures in the paper, e.g., sway and pitch.  Regarding the blade frequency, it may be easier to see this if you ensemble average the results to eliminate the noise in your RAOs.

You can independently set the wind and wave directions in OpenFAST via `PropagationDir` in InflowWind for the wind direction and `WaveDir` in HydroDyn for the wave direction.  Wind and waves will be aligned when `PropagationDir = -WaveDir`.  (Please note that `PropagationDir` has the opposite sign convention as `WaveDir`.)

Typically, `WaveHs` and `WaveTp` are chosen based on the site-specific conditions you are trying to simulate.  Typically, larger `WaveHs` correspond with larger `WaveTp`.  Setting too small of `WaveTp` for a given `WaveHs` may result in unrealistically steep waves.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
Thank you for your reply. I'd like to ask you a more specific method, how to enable average the results to eliminate the noise? Although this question is stupid, I'm not sure how to set it. For example: time=Data(:,1);U=Data(:,2);Y=Data(:,3);nens=6; If the results obtained by setting nens=4/6 do not meet our expectations, then whether the nens can be adjusted freely by ourselves until the expected results appear？

How to set specifically Mooring line loss and Flooded column working condition in OC4?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Yes, you can set `nens` to different values.   But the higher the value of `nens`, the shorter each ensemble will be, so, the coarser your RAO will be.  If you want to increase `nens` a lot, I would suggest using a long time series.

Simulating a mooring line loss requires a simple source code change, as documented in the following topic on our forum: https://forums.nrel.gov/t/tendon-failure-in-fast-modeling/934.

The flooded column load case in OC4 was ran by changing input parameter `FillFSLoc` in the HydroDyn input file.

Best regards,
